### PR TITLE
doc: Upgrade instructions from 1.3 to 1.4

### PR DESCRIPTION
**Pull request by mateusoliveira43**
## Why the changes were made

Add steps to upgrade from OADP 1.3 to 1.4.

## How to test the changes made

Check if all changes were documented.

Check that all necessary changes were documented.

Test an upgrade following [upgrade test documentation](https://github.com/openshift/oadp-operator/blob/master/docs/developer/testing/test_oadp_version_upgrade.md).

### CODE REVIEW

File: docs/upgrade_1-3_to_1-4.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+- Velero was updated from version 1.12 to 1.14 (Changes reference: https://velero.io/docs/v1.13/upgrade-to-1.13/ https://velero.io/docs/v1.14/upgrade-to-1.14/)
+
+    From this update:
+
+    - velero-plugin-for-csi code is now inside Velero code, which means, no init container is needed for the plugin anymore. No changes needed in DPA.
+
+    - Velero changed client Burst and QPS defaults from 30 and 20 to 100 and 100, respectively.
+
+    - velero-plugin-for-aws updated default value of `spec.config.checksumAlgorithm` field in BackupStorageLocations (BSLs) from `""` (no checksum calculation) to `CRC32` (reference https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/release-1.10/backupstoragelocation.md). For compatibility, OADP did not change default value of the field for BSLs created within DPA (and if you want to change it, use `spec.backupLocations[].velero.config.checksumAlgorithm` field). If your BSLs are created outside DPA, you may need to change value of the `spec.config.checksumAlgorithm` field in the BSLs. Some compatible S3 storages, like IBM, currently only work with `""` value.
```
Comment (weshayutin) [inline docs/upgrade_1-3_to_1-4.md:15]: change:
For compatibility, OADP did not change default value of the field for BSLs created within DPA (and if you want to change it, use `spec.backupLocations[].velero.config.checksumAlgorithm` field).
to:
The checksum algorithm types are known to work AWS only.  Please confirm with your storage provider with regards to the md5sum algorithm support and configuration.   The OADP configuration can be updated by setting the value `spec.config.checksumAlgorithm` in the DPA.  The OADP default configuration  is set to `""`, meaning the md5sum is not checked.  Several S3 providers require the md5sum set to `""`.   The checksumAlgorithm setting of `""` is consistent with OADP-1.3.x.

File: docs/upgrade_1-3_to_1-4.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+### Upgrade the OADP Operator
+
+For general operator upgrade instructions please review the [OpenShift documentation](https://docs.openshift.com/container-platform/4.13/operators/admin/olm-upgrading-operators.html)
+* Change the Subscription for the OADP Operator from `stable-1.3` to `stable-1.4`
+* Allow time for the operator and containers to update and restart
+
+### Convert your DPA to the new version
+
+No changes.
```
Comment (weshayutin) [inline docs/upgrade_1-3_to_1-4.md:36]: little more verbose, perhaps.  No DPA changes are required to upgrade from OADP 1.3 to OADP 1.4

Comment (mateusoliveira43) [reply docs/upgrade_1-3_to_1-4.md:36]: done

File: docs/upgrade_1-3_to_1-4.md (hunk 1 above)
Comment (mateusoliveira43) [reply docs/upgrade_1-3_to_1-4.md]: updated

Can you check if text is ok now?

File: docs/upgrade_1-3_to_1-4.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+### Upgrade the OADP Operator
+
+For general operator upgrade instructions please review the [OpenShift documentation](https://docs.openshift.com/container-platform/4.13/operators/admin/olm-upgrading-operators.html)
+* Change the Subscription for the OADP Operator from `stable-1.3` to `stable-1.4`
+* Allow time for the operator and containers to update and restart
+
+### Convert your DPA to the new version
+
+No DPA changes are required to upgrade from OADP 1.3 to 1.4.
```
Comment (amastbau) [inline docs/upgrade_1-3_to_1-4.md:38]: what about the csi plugin,  or this is only for api changes?

**Review by weshayutin — DISMISSED** [review]
/LGTM

**Review by weshayutin — APPROVED** [review]
/LGTM

**Pull request merged**